### ISSUE TITLE
Printing of macro symbols displays like a macrocall

### ISSUE THREAD

**Issue by mbauman** [JuliaLang/julia#32354]
MWE:

```julia
julia> @macroexpand @which @which
:(InteractiveUtils.which(@which, Tuple{LineNumberNode,Module}))

julia> dump(@macroexpand @which @which)
Expr
  head: Symbol call
  args: Array{Any}((3,))
    1: GlobalRef
      mod: Module InteractiveUtils
      name: Symbol which
    2: Symbol @which
    3: Tuple{LineNumberNode,Module} <: Any
```

This should protect the macro name and print it as a `Symbol("@which")`.

**Comment by JeffBezanson** [comment]
Looks like a more general problem:
```
julia> Symbol("##")
Symbol("##")

julia> Expr(:tuple, Symbol("##"))
:((##,))
```
So there's some code for this, but it's not used inside expressions.

**Comment by c42f** [comment]
More entertainment:

```
julia> Expr(:tuple, Symbol("a-b"))
:((a-b,))
```

**Comment by c42f** [comment]
So I've got a fix for this over at #23408, but It's worth noting that it dramatically uglifies printing of symbols generated by macro hygiene, even as it makes the resulting `Expr`s valid.

```julia
julia> macro ugly()
           quote
               x = 1
               y = 2
           end
       end
@ugly (macro with 1 method)

julia> @macroexpand @ugly
quote
    #= REPL[30]:3 =#
    $(Symbol("#57#x")) = 1
    #= REPL[30]:4 =#
    $(Symbol("#58#y")) = 2
end
```

Of course, for non-trivial macros it's a lot worse.

**Comment by c42f** [comment]
On the up side, the result of printing such expanded forms is actually valid code with those changes.
